src/main/java/ua/com/danit/controller/CommentController.java

```diff
@@ -23,20 +25,20 @@ public CommentController(CommentService commentService) {
   }
 
   @PostMapping("/{postId}")
-  public ResponseEntity<Comment> addComment(@PathVariable String postId, @RequestBody Comment comment) {
-    commentService.addComment(Long.parseLong(postId), comment.getText());
-    return ResponseEntity.ok(comment);
+  public ResponseEntity<Comment> createComment(@PathVariable String postId, @RequestBody Comment comment) {
+    Comment resComment = commentService.createComment(Long.parseLong(postId), comment.getText());
+    return ResponseEntity.ok(resComment);
   }
 
   @PutMapping("/{commentId}")
   public ResponseEntity<Comment> updateComment(@PathVariable String commentId, @RequestBody Comment comment) {
-    commentService.updateComment(Long.parseLong(commentId), comment.getText());
-    return ResponseEntity.ok(comment);
+    Optional<Comment> resComment = commentService.updateComment(Long.parseLong(commentId), comment.getText());
+    return resComment.map(ResponseEntity::ok).orElseGet(() -> (ResponseEntity<Comment>) ResponseEntity.notFound());
   }
 
   @DeleteMapping("/{commentId}")
   public ResponseEntity<Comment> deleteComment(@PathVariable String commentId) {
-    commentService.deleteComment(Long.parseLong(commentId));
-    return ResponseEntity.ok(new Comment());
+    Optional<Comment> resComment = commentService.deleteComment(Long.parseLong(commentId));
+    return resComment.map(ResponseEntity::ok).orElseGet(() -> (ResponseEntity<Comment>) ResponseEntity.notFound());
```

Comment:
I don't know what to return here

Reply:
Response entity, containing deleted comment. If you throw an exception in service as I mentioned earlier - you won't have problem with optional